### PR TITLE
[#349] Add Twilio source documentation

### LINKED ISSUE #349: Twilio source documentation
Organize under Sms and Whatsapp so that people interested in those concerns can more easily find it.

### PR DESCRIPTION
**Pull request by chrismatix**
fixes #349

### CODE REVIEW

File: docs/docs/sources/sms-twilio.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+Next call the Platform API:
+
+```
+POST /channels.connect
+```
+
+- `source` *must* be `sms.twilio`
+- `source_channel_id`   The phone number as listed in your [Twilio dashboard](https://www.twilio.com/console/phone-numbers/). 
+                        Must not have spaces and must include the country code.
```
Comment (paulodiniz) [inline docs/docs/sources/sms-twilio.md:38]: Is this big space a github issue? I guess

File: docs/docs/sources/whatsapp-twilio.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+TWILIO_AUTH_TOKEN=<YOUR AUTH TOKEN>
+```
+
+You also have to add your account sid:
+
+```
+TWILIO_SID=<YOUR ACCOUNT SID>
+```
+
+
```
Comment (paulodiniz) [inline docs/docs/sources/whatsapp-twilio.md:26]: Extra line

**Review by paulodiniz — APPROVED** [review]

File: docs/docs/sources/sms-twilio.md (hunk 1 above)
Comment (chrismatix) [reply docs/docs/sources/sms-twilio.md:38]: Oh this is funny. In my editor this shows up tabbed 😅  I even checked the docusaurus output and there it's all one line that wraps

File: docs/docs/sources/sms-twilio.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+- `source` *must* be `sms.twilio`
+- `source_channel_id`   The phone number as listed in your [Twilio dashboard](https://www.twilio.com/console/phone-numbers/). 
+                        Must not have spaces and must include the country code. 
+- `token` leave empty.
+
+```json5
+{
+  "source": "sms.twilio",
+  "source_channel_id": "+491234567",
+  "name": "My custom name for this location",
```
Comment (lucapette) [inline docs/docs/sources/sms-twilio.md:45]: not sure what's better than location here...

File: docs/docs/sources/whatsapp-twilio.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
+---
+title: Whatsapp with Twilio
```
Comment (lucapette) [inline docs/docs/sources/whatsapp-twilio.md:2]:
```suggestion
title: Whatsapp via Twilio
```

File: docs/docs/sources/sms-twilio.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
+---
+title: SMS with Twilio
+sidebar_label: SMS - Twilio
+---
+
+The Twilio sms source provides a channel for sending and receiving SMS using the [Twilio API](https://www.twilio.com/).
+
+This document assumes that you have already created a Twilio account.
```
Comment (lucapette) [inline docs/docs/sources/sms-twilio.md:8]:
```suggestion
This document assumes that you have a Twilio account.
```

File: docs/docs/sources/sms-twilio.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
+---
+title: SMS with Twilio
```
Comment (lucapette) [inline docs/docs/sources/sms-twilio.md:2]:
```suggestion
title: SMS via Twilio
```

File: docs/docs/sources/whatsapp-twilio.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+sidebar_label: Whatsapp - Twilio
+---
+
+The Twilio Whatsapp source provides a channel for sending and receiving Whatsapp messages
+using the [Twilio API](https://www.twilio.com/).
+
+This document assumes that you have already created a Twilio account and connected it to [Whatsapp](https://www.twilio.com/whatsapp).
```
Comment (lucapette) [inline docs/docs/sources/whatsapp-twilio.md:9]:
```suggestion
This document assumes that you have a Twilio account connected to [Whatsapp](https://www.twilio.com/whatsapp).
```

File: docs/docs/sources/whatsapp-twilio.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+```
+TWILIO_AUTH_TOKEN=<YOUR AUTH TOKEN>
+```
+
+You also have to add your account sid:
+
+```
+TWILIO_SID=<YOUR ACCOUNT SID>
+```
```
Comment (lucapette) [inline docs/docs/sources/whatsapp-twilio.md:24]:
```suggestion
You must create a [Twilio auth token](https://support.twilio.com/hc/en-us/articles/223136027-Auth-Tokens-and-How-to-Change-Them) and add it to `infrastructure/airy.conf` together with your account sid:

```
TWILIO_AUTH_TOKEN=<YOUR AUTH TOKEN>
TWILIO_SID=<YOUR ACCOUNT SID>
```

File: docs/docs/sources/sms-twilio.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+```
+TWILIO_AUTH_TOKEN=<YOUR AUTH TOKEN>
+```
+
+You also have to add your account sid:
+
+```
+TWILIO_SID=<YOUR ACCOUNT SID>
+```
```
Comment (lucapette) [inline docs/docs/sources/sms-twilio.md:23]: I suggested a way to rewrite this so it's a little lighter, I guess we could reuse the markdown 

As I understand it, docusaurus supports it via https://mdxjs.com/getting-started#documents

File: docs/docs/sources/whatsapp-twilio.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+
+## Connecting a channel
+
+After you have created a Twilio phone number you must [point its webhook](https://www.twilio.com/docs/sms/tutorials/how-to-receive-and-reply-java#configure-your-webhook-url) to your
+Airy Core installation.
```
Comment (lucapette) [inline docs/docs/sources/whatsapp-twilio.md:30]:
```suggestion
After you created a Twilio phone number, you must [point its webhook](https://www.twilio.com/docs/sms/tutorials/how-to-receive-and-reply-java#configure-your-webhook-url) to your Airy Core Platform running instance.
```

(been using "running instance" based on some documentation @paulodiniz wrote)

File: docs/docs/sources/whatsapp-twilio.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+Next call the Platform API:
```
Comment (lucapette) [inline docs/docs/sources/whatsapp-twilio.md:32]:
```suggestion
Next call the Airy Core Platform API for connecting channels:
```

File: docs/docs/sources/whatsapp-twilio.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+```
+POST /channels.connect
+```
+
+- `source` *must* be `whatsapp.twilio`
+- `source_channel_id`   The phone number as listed in your [Twilio dashboard](https://www.twilio.com/console/phone-numbers/). 
+                        Must not have spaces, must include the country code and be prefixed by `whatsapp:`
```
Comment (lucapette) [inline docs/docs/sources/whatsapp-twilio.md:40]:
```suggestion
                        Must not have spaces, must include the country code, and be prefixed by `whatsapp:` 
```

**Review by lucapette — CHANGES_REQUESTED** [review]
great work!

I'm requesting changes as per usual little suggestions. About the re-use of the configuration block, I would do already here (but it would be fine with a subsequent issue)

File: docs/docs/sources/whatsapp-twilio.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+The Twilio Whatsapp source provides a channel for sending and receiving Whatsapp messages
+using the [Twilio API](https://www.twilio.com/).
+
+This document assumes that you have a Twilio account connected to [Whatsapp](https://www.twilio.com/whatsapp).
+
+## Configuration
+ 
+import TwilioSource from './twilio-source.mdx'
+
+<TwilioSource />
```
Comment (chrismatix) [inline docs/docs/sources/whatsapp-twilio.md:15]: monkey

File: docs/docs/sources/sms-twilio.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+## Configuration
+ 
+import TwilioSource from './twilio-source.mdx'
+
+<TwilioSource />
+
+## Connecting a channel
+
+After you have created a Twilio phone number you must [point its webhook](https://www.twilio.com/docs/sms/tutorials/how-to-receive-and-reply-java#configure-your-webhook-url) to your
+Airy Core installation.
```
Comment (lucapette) [inline docs/docs/sources/sms-twilio.md:19]:
```suggestion
After you created a Twilio phone number you must [point its webhook](https://www.twilio.com/docs/sms/tutorials/how-to-receive-and-reply-java#configure-your-webhook-url) to your
Airy Core Platform running instance.
```

**Review by lucapette — APPROVED** [review]
:shipit:

**Pull request merged**